### DISCUSSION TITLE
Pipelineruns with `PipelineRunPending` state are not timing out

### DISCUSSION THREAD

**Discussion by VeereshAradhya** [tektoncd/pipeline#3830] [General]
I was testing PipelineRunPending state and I observed that when I set the timeout even after the timeout exceeds the pipelinerun stays in Running(PipelineRunPending) status. Is it because the pipelinerun is not started yet? Is this expected?

Below are the pipelinerun which are in `Running(PipelineRunPending)` state from more than 2 days
```
$ tkn pr ls
NAME                         STARTED      DURATION     STATUS
new-pipeline-0qajk12         ---          ---          Running(PipelineRunPending)
new-pipeline-12345           ---          ---          Running(PipelineRunPending)
new-pipeline-timeout-check   ---          ---          Running(PipelineRunPending)
```

**Comment by afrittoli** [comment]
This is expected behaviour, see the corresponding [TEP-0015](https://github.com/tektoncd/community/blob/main/teps/0015-pending-pipeline.md). Having a timeout for pipelines in pending state was not part of the original feature design. The responsibility for starting pipelines in pending state lies outside of the Tekton controller, so it could be argued that the same applies for timeout or cleanup of pipeline runs that are in pending state.
